ocs_ci/utility/kms.py

```diff
@@ -761,36 +764,304 @@ def create_vault_csi_kms_token(
+            vault_backend (str): KV version to be used, either kv or kv-v2
+            vault_backend_path (str): The backend path in Vault where the encryption
+                                      keys will be stored
+            vault_namespace (str): Namespace in Vault, if exists, where the backend
+                                   path is created
+            vault_role (str): (Vaulttenantsa) The role name in Vault configured with
+                              kube auth method for the given policy and tenant namespace
+            vault_auth_path (str): (Vaulttenantsa) The path where kubernetes auth
+                                   method is enabled
+            vault_auth_namespace (str): (Vaulttenantsa) The namespace where kubernetes
+                                        auth method is enabled, if exists
+        """
+
+        logger.info(f"Creating tenant configmap in namespace {tenant_namespace}")
+        tenant_cm = templating.load_yaml(constants.RBD_CSI_VAULT_TENANT_CONFIGMAP)
+        tenant_cm["metadata"]["namespace"] = tenant_namespace
+
+        if vault_backend is None:
+            tenant_cm["data"].pop("vaultBackend")
+        else:
+            tenant_cm["data"]["vaultBackend"] = vault_backend
+
+        if vault_backend_path is None:
+            tenant_cm["data"].pop("vaultBackendPath")
+        else:
+            tenant_cm["data"]["vaultBackendPath"] = vault_backend_path
+
+        if vault_namespace is None:
+            tenant_cm["data"].pop("vaultNamespace")
+        else:
+            tenant_cm["data"]["vaultNamespace"] = vault_namespace
+
+        if vault_role is None:
+            tenant_cm["data"].pop("vaultRole")
+        else:
+            tenant_cm["data"]["vaultRole"] = vault_role
+
+        if vault_auth_path is None:
+            tenant_cm["data"].pop("vaultAuthPath")
+        else:
+            tenant_cm["data"]["vaultAuthPath"] = vault_auth_path
+
+        if vault_auth_namespace is None:
+            tenant_cm["data"].pop("vaultAuthNamespace")
+        else:
+            tenant_cm["data"]["vaultAuthNamespace"] = vault_auth_namespace
+
+        self.create_resource(tenant_cm, prefix="tenant-cm")
+        logger.info("Tenant ConfigMap ceph-csi-kms-config created successfully")
+
+    def vault_kube_auth_setup(
+        self,
+        auth_path=None,
+        auth_namespace=None,
+        token_reviewer_name=constants.RBD_CSI_VAULT_TOKEN_REVIEWER_NAME,
+    ):
+        """
+        Setup kubernetes auth method in Vault
+
+        Args:
+            auth_path (str): The path where kubernetes auth is to be enabled.
+                             If not provided default 'kubernetes' path is used
+            auth_namespace (str): The vault namespace where kubernetes auth is
+                                  to be enabled, if applicable
+            token_reviewer_name (str): Name of the token-reviewer serviceaccount
+                                       in openshift-storage namespace
+
+        """
```

Comment:
Please add `Raises: NotFounderror` in the doc section

Reply:
Done